### PR TITLE
Update disabled state logic for v2

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
v1 and v2 of the functions runtime disable functions in different ways. Update logic to reflect that. And clean up code a bit - I think this was originally written before `refreshImpl` or `description` existed

Fixes https://github.com/Microsoft/vscode-azurefunctions/issues/780

### CODE REVIEW

File: src/tree/FunctionTreeItem.ts [hunk 1]
```diff
@@ -107,6 +116,40 @@ export class FunctionTreeItem extends AzureTreeItem<ISiteTreeRoot> {
...
+        if (projectRuntime === ProjectRuntime.v1) {
+            this._disabled = this.config.disabled;
+        } else {
+            const key: string = `AzureWebJobs.${this._name}.Disabled`;
+            /**
+             * The docs only officially mentioned 'true' and 'false', but here is what I found:
+             * The following resulted in a disabled function:
+             * true, tRue, 1
+             * The following resulted in an enabled function:
+             * false, fAlse, 0, fdsaf, 2, undefined
```
Comment (nturinski) [inline src/tree/FunctionTreeItem.ts]: Did somebody have a stroke while typing this?

Comment (EricJizbaMSFT) [reply src/tree/FunctionTreeItem.ts]: Lol what problem do you have with it?

Comment (EricJizbaMSFT) [reply src/tree/FunctionTreeItem.ts]: :) Exactly gotta test some edge cases - specifically a random word other than 'true' or 'false'
